src/core/server/ui_settings/settings/notifications.ts

```diff
@@ -96,15 +84,11 @@ export const getNotificationsSettings = (): Record<string, UiSettingsParams> =>
       value: 5000,
       description: i18n.translate('core.ui_settings.params.notifications.infoLifetimeText', {
         defaultMessage:
-          'The time in milliseconds which an information notification will be displayed on-screen for. ' +
-          'Setting to {infinityValue} will disable.',
-        values: {
-          infinityValue: 'Infinity',
-        },
+          'The time in milliseconds which an information notification will be displayed on-screen for. ',
       }),
       type: 'number',
       category: ['notifications'],
-      schema: schema.oneOf([schema.number({ min: 0 }), schema.literal('Infinity')]),
+      schema: schema.number({ min: 0 }),
```

Comment:
This is a breaking change for customers, as any Kibana instance currently overriding this value via the `uiSettings.overrides` config property will no longer start due to a config validation error.

This will also cause warnings every time the uiSettings are fetched:

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/server/ui_settings/ui_settings_client.ts#L174-L181

If we really want to remove this value from these settings, our only option would be to add the associated migration in `src/core/server/ui_settings/saved_objects/migrations.ts` to migrate the value to a new valid one.

However, I'm not sure we should really drive a model change only based on a UI limitation?

I would personally have kept that, and just update the UI part to no longer document that `Infinity` is a possible override from the UI.

Reply:
> I would pragmatically use this solution for now though, as developing a custom component (and adding the associated uiSettings type) seems to much work for the gain here

I agree, and will take the first approach.

Reply:
Not sure, we may just not have any documentation on it tbh. Greping `uiSettings.overrides` in `/docs` has no match... @mshustov maybe you know?

Reply:
they are not documented in https://www.elastic.co/guide/en/kibana/current/advanced-options.html nor in https://www.elastic.co/guide/en/kibana/current/settings.html

I'd ask Docs team whether we want to announce this API to the wider audience.

Reply:
> we may just not have any documentation on it tbh

Yeah, AFAIK this has always been an undocumented feature since [it was added in 6.5](https://github.com/elastic/kibana/pull/21628). It was in our release notes, but I don't think it was ever added to the documentation.

Reply:
For now, I'll follow up with a docs PR to remove the sentences mentioning 'Infinity' as an options for the [notifications lifetime settings in Advanced Settings](https://www.elastic.co/guide/en/kibana/current/advanced-options.html#kibana-notification-settings).

Reply:
@elastic/kibana-core it would be nice to have this in for 7.13. If someone has time to have another look, that would be great 😄